opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -503,9 +500,130 @@ message DoubleHistogramDataPoint {
   // a boolean value which decides what type of intervals to use.
   repeated double explicit_bounds = 7;
 
+  // Multiple ways to define the bound sequence is supported. While explicit_bounds is one of the ways,
+  // for backward compatibility reason, it is not included in "oneof bound_type".
+  // Only one of explicit_bounds or bound_type should be present. When both are present, bound_type is ignored.
+  oneof bound_type {
+    LinearBounds linear_bounds = 9;
+    ExponentialBounds exponential_bounds = 10;
+    LogLinearBounds log_linear_bounds = 11;
+  }
   // (Optional) List of exemplars collected from
   // measurements that were used to form the data point
   repeated DoubleExemplar exemplars = 8;
+
+  // LinearBounds. Bounds can be generated via
+  // for (int i = 0; i < num_of_bounds; i++)
+  //     boundList.add(offset + width * i);
+  message LinearBounds {
+    double offset = 1;
+    double width = 2;
+    uint32 num_of_bounds = 3;
+  }
+
+  // ExponentialBounds. Bounds are on log scale.
+  // The bound sequence is stitched together using negative bounds, zero bound, and positive bounds.
+  // The general rules is: bound = reference * power(base, exponent)
+  // Reference and base are constants. Exponent is integer index of histogram buckets.
+  // Reference defaults to 1. Base should be greater than 1.
+  // Negative index is allowed. It results in a bound between 0 and the reference.
+  // Two sections are specified, one for positive numbers, one for negative numbers.
+  // A special field is used to specify if a bound of zero exists.
+  // The formula to generate explicit bounds here are for demonstration purpose only.
+  // If the message receiver natively stores exponential histograms, it will not need to generate explicit bounds.
+  // It will only need to record the exponential bound parameters.
+  message ExponentialBounds {
```

Comment:
This histogram bound type represents buckets of data points with finer granularity close to a central point, and less granularity as the bucket range gets farther from that central point. Like this:
```
                                    0
|________________|________|____|__|_|_|__|____|________|________________|
```

I find it surprising, though, that this central point is hardcoded to zero. This is likely useful for values that are clustered close to zero, and that may be positive or negative. But what about values that cluster around some very large number? 
Instead of zero, I think I would have expected this inflection point to be something like the median of the data.

From the message perspective, I think the only thing required to support this would be to add an `offset` field. Then the bound would be computed like:
```
reference * power(base, exponent) + offset
```

I don't know what implications this has for downstream aggregators of these histograms. Would it be problematic to merge histrogams whose center values were different?
Like this:
```
                  100
                   |
|________|____|__|_|_|__|____|________|

                             110
                              |
           |________|____|__|_|_|__|____|________|
```

Reply:
Applying a linear offset to an exponential series is possible. 
For the range of 50 to 150, a series with offset of 100 looks like:
```
                  100
                   |
|________|____|__|_|_|__|____|________|
50                                   150

```
The same range, with offset 0, looks like:

```
|___|____|_____|_____|_______|________|
50                                   150

```
Which form would be more generally useful? I think 0 offset is. Allowing variable offset would add more complexity. If there is a use case, a user can always apply the offset before inserting data to the histogram. For example, insert(x - 100) to achieve the same effect. Of course, on reading the histogram, the user needs to add 100 back. 

As you suggested, arbitrary offset would make merge difficult, because bounds are not aligned.

Now onto positive and negative numbers. For simplicity, the protocol here requires the same base and reference for positive and negative series (DDSketch also does this). Thus the bound space is always symmetrical like:
```
                                    0
|________________|________|____|__|_|_|__|____|________|________________|
NOT like 
                                                   0
|________________________|________________|________|_|__|____|________|________________|

```The separate positive and negative sections define index ranges where the bounds are actually present in the infinite logical range. Example:
```
                                    0
|________________|________|____|__|_|_|__|____|________|________________|
                   |---------|        |---------------------------------------------|
                  -100     -40        +5                                          +320
                 negative range       positive range

```

For negative numbers and positive numbers of the same absolute value, the bucket granularity is identical (symmetrical with reflection point at zero). But we allow asymmetrical value ranges of negative and positive numbers. 
Combining the two ranges into one is not a good idea because there are an infinite number of indexes between 0 and 1 (assuming reference of 1) for infinite precision numbers. For finite precision numbers such as 64 bit double, there is still a large number of indexes. For a base of 2, the number is around 1024. Including all these empty buckets in the series is not good.
Theoretically, we could apply a linear offset to shift all input numbers to the positive range, for bounds like
```
 |___|____|_____|_____|_______|________|
-50                                   150

```
Though in some cases, this may be preferred (like temperature in Celsius or Fahrenheit) . But for the common cases we care, like response time, bank balance, who all have natural zero point, symmetric buckets around 0 are preferred. Again, for special cases like temperature, users can shift before inserting to histogram, or use a more scientific unit like Kelvin.
And there is also a practical reason for not shifting the range, because the range is often unknown in streaming data. Range shifting adds complexity to histogram management.